File: Backend/SharedLib/Model/PaginationModel.cs
namespace SharedLib.Model
{
    public class PaginationRequestModel
    {
        public int CurrentPage { get; set; }
        public int PageSize { get; set; }
        public string Keyword { get; set; } = string.Empty;
        public string SortField { get; set; } = string.Empty;
        public string SortOrder { get; set; } = string.Empty;
        public IEnumerable<PaginationFilterModel> Filters { get; set; } = [];
    }

    public class PaginationFilterModel
    {
        public string FieldName { get; set; } = string.Empty;
        public string FieldValue { get; set; } = string.Empty;
    }

    public class PaginationModel<T> where T : class
    {
        public int TotalRecord { get; set; }
        public int TotalPage
        {
            get => PageSize == 0 ? 0 : (int)Math.Ceiling((double)TotalRecord / PageSize);
        }
        public int CurrentPage { get; set; }
        public int PageSize { get; set; }
        public IEnumerable<T> Data { get; set; } = [];
    }
}

File: Frontend/lms-public/src/util/types/paginationModel.ts
export class PaginationModel<T> {
  constructor() {
    this.totalRecord = 0;
    this.currentPage = 1;
    this.pageSize = 10;
    this.data = [] as T[];
  }
  totalRecord: number;
  currentPage: number;
  pageSize: number;
  data: T[];
  get totalPage(): number {
    return Math.ceil(this.totalRecord / this.pageSize);
  }
}
